zmq.hpp

```diff
@@ -351,16 +351,19 @@ inline int poll(std::vector<zmq_pollitem_t> const &items, long timeout_ = -1)
     return detail::poll(const_cast<zmq_pollitem_t *>(items.data()), items.size(), timeout_);
 }
 
-inline int
-poll(zmq_pollitem_t *items, size_t nitems, std::chrono::milliseconds timeout = std::chrono::milliseconds{-1})
+template<typename Duration = std::chrono::milliseconds>
+int
+poll(zmq_pollitem_t *items, size_t nitems, Duration timeout = std::chrono::milliseconds{-1})
 {
-    return detail::poll(items, nitems, static_cast<long>(timeout.count()));
+    auto timeout_ms = std::chrono::duration_cast<std::chrono::milliseconds>(timeout);
+    return detail::poll(items, nitems, static_cast<long>(timeout_ms.count()));
```

Comment:
One concern is that for poll and poller, -1 millisecond is a special value that may need special handling. E.g. does -1s or -1ns do the right thing?

> If the value of timeout is -1, zmq_poll() shall block indefinitely until a requested event has occurred on at least one zmq_pollitem_t.

Reply:
Yes, but you cast the input Duration to milliseconds then pass that to libzmq.